baybe/acquisition/base.py

```diff
@@ -0,0 +1,71 @@
+"""Base classes for all acquisition functions."""
+
+from __future__ import annotations
+
+import warnings
+from abc import ABC
+from typing import ClassVar, Union
+
+from attrs import define
+
+from baybe.acquisition.adapter import debotorchize
+from baybe.serialization.core import (
+    converter,
+    get_base_structure_hook,
+    unstructure_base,
+)
+from baybe.serialization.mixin import SerialMixin
+from baybe.surrogates.base import Surrogate
+
+
+@define(frozen=True)
+class AcquisitionFunction(ABC, SerialMixin):
+    """Abstract base class for all acquisition functions."""
+
+    _abbreviation: ClassVar[str]
+    """An alternative name for type resolution."""
+
+    def to_botorch(self, surrogate: Surrogate, best_f: float):
+        """Create the botorch-ready representation of the function."""
+        import botorch.acquisition as botorch_acqf
+
+        acqf_cls = getattr(botorch_acqf, self.__class__.__name__)
+
+        return debotorchize(acqf_cls)(surrogate, best_f)
+
+
+# Register de-/serialization hooks
+def _add_deprecation_hook(hook):
+    """Adjust the structuring hook such that it auto-fills missing target types.
+
+    Used for backward compatibility only and will be removed in future versions.
+    """
+
+    def added_deprecation_hook(val: Union[dict, str], cls):
```

Comment:
ok seems the `_add_deprecation_hook` did only care about the string case and not for dict case where the type key could have been in th `type` field

Having said that, this is strictly speaking not a deprecation. 

No one can encoutner this situation as the specification of acqfs via dict and `type` did not exist before, so should we even include this in deprecations? I think this though crossed my mind during working on this PR and hence that was not included

Reply:
nvm I simply extended the deprecation and the test, not sure about the `    converter.structure(
        {
            "type": "SequentialGreedyRecommender",
            "acqf": t,
        },
        RecommenderProtocol,
    )` part needed, I took only the last line from your snippet and added it to the existing test, please resolve if sufficient

Reply:
Ah, I guess you have a point here, the dict version could not have occurred before, so technically no deprecation mechanism needed 👍🏼. But OK, now you solved it anyway, so I think that's just fine. Thanks for spending the (unnecessary) effort! 😄 my bad ...

Regarding the upper code snippet: sorry, I think that was a leftover from me playing around, but pretty much confirms that the above is no deprecation, because it implements the old way of specifying the acqf, which passes without problems 👍🏼 

Feel free to resolve, unless further discussion needed.